Tests/KituraTests/TestResponse.swift

```diff
@@ -1481,6 +1498,22 @@ class TestResponse: KituraTest {
             XCTAssert(response.statusCode == .notFound)
         }
 
+        
+        router.get("user_info", handler: {
+            _, response, next in
+            // Store something in userInfo
+            response.userInfo["greeting"] = "hello"
+            next()
+        }, {
+            _, response, next in
+            // Read the value that should be stored in userInfo
+            guard let greeting = response.userInfo["greeting"] as? String else {
```

Comment:
This relies on the handlers being run by Kitura in the order that they are defined. Although we don't explicitly document this behaviour, it is the same as Express which also runs handlers in order. As such, we are unlikely to change this even though technically it is relying on undefined behaviour. A roundabout way of saying this is probably OK!

Reply:
Ah OK, I didn't know this was undefined and assumed they were called in the order they are defined. Isn't this also why a global error handler needs to be registered last?

Reply:
@ianpartridge I was under the impression that we did rely on the handlers being invoked in definition order, for example to ensure that middlewares run (eg. session) before handlers that depend on them.  We should definitely document it.